src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,34 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+    public static String getSalaryInfo(String[] names, String[] data,
+                                       String dateFrom, String dateTo) {
+        LocalDate localDateFrom = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate localDateTo = LocalDate.parse(dateTo, FORMATTER);
+
+        String[] arrDates;
+        StringBuilder result = new StringBuilder().append("Report for period ")
+                .append(dateFrom).append(" - ").append(dateTo);
+        for (String name : names) {
+
+            int salary = 0;
+            for (String dates : data) {
+                arrDates = dates.split(" ");
+                LocalDate dateFromData = LocalDate.parse(arrDates[0], FORMATTER);
+                if (localDateFrom.compareTo(dateFromData) <= 0
+                        && localDateTo.compareTo(dateFromData) >= 0
+                        && name.equals(arrDates[1])) {
+                    salary += (Integer.parseInt(arrDates[2])
+                            * Integer.parseInt(arrDates[3]));
```

Comment:
create constants for indexes

Reply:
I've never seen constants created for indexes, and it's not clear what this does except obstruct the code, but maybe that's the point.

Reply:
not in this case. when parsing some string, when you split it, after a few days you would still see, what number is on which position